packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -1217,7 +1217,6 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
         _obscureLatestCharIndex = _value.selection.baseOffset;
       }
     }
-    _lastKnownRemoteTextEditingValue = value;
```

Comment:
This is the key piece here - we update this value in the call path of `_formatAndSetValue` anyway, and if we update it here, a later check that compares the value being passed with `_lastKnownRemoteTextEditingValue` always returns true.

This is just too early to set the value -we haven't actually notified the remote side yet.

Reply:
SGTM